javascript/app-framework/src/components/app/DataShaperApp/DataShaperApp.constants.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { KnownProfile } from '@datashaper/schema'
+import type { DataTable } from '@datashaper/workflow'
+
+import type { ProfileHandlerPlugin} from '../../../types.js';
+import { ResourceGroup } from '../../../types.js'
+import {
+	BundleEditor,
+	CodebookEditor,
+	DataSourceEditor,
+	RawTableViewer,
+	WorkflowEditor,
+} from '../../editors/index.js'
+
+export const KNOWN_PROFILE_PLUGINS: ProfileHandlerPlugin[] = [
+	{
+		profile: KnownProfile.TableBundle,
+		renderer: BundleEditor,
+		iconName: 'ViewAll',
+		group: ResourceGroup.Data,
+	},
+	{
+		profile: KnownProfile.DataTable,
+		renderer: DataSourceEditor,
+		iconName: 'PageData',
+		group: ResourceGroup.Data,
```

Comment:
Yeah, this makes sense - we invert that control and publish in the API what options the resource can opt into